activesupport/lib/active_support/json/encoding.rb

```diff
@@ -91,14 +81,14 @@ def to_s
           # calls.
           def jsonify(value)
             case value
-            when String
-              EscapedString.new(value)
-            when Numeric, NilClass, TrueClass, FalseClass
+            when String, Integer, Symbol, nil, true, false
+              value
+            when Numeric
               value.as_json
             when Hash
               result = {}
               value.each do |k, v|
-                result[jsonify(k)] = jsonify(v)
+                result[keyify(k)] = jsonify(v)
```

Comment:
```suggestion
                k = k.to_s unless Symbol === k || String === k
                result[k] = jsonify(v)
```

Do you think it may be worth inlining `keyify`?

Reply:
It's a good idea. I'm now also wondering if we should could just leave the keys unchanged (regardless of their class) and expect that JSON will `to_s` them (unlike values, our behaviour matches JSON's, we just want `.to_s`). I'll test that out in a follow up PR though.